src/renderers/shared/reconciler/__tests__/refs-destruction-test.js

```diff
@@ -59,4 +59,52 @@ describe('refs-destruction', function() {
     ReactDOM.render(<TestComponent destroy={true} />, container);
     expect(Object.keys(testInstance.refs || {}).length).toEqual(0);
   });
+
+  it('should not error when destroying child with ref asynchronously', function() {
+    var flag = false;
+    runs(function() {
+      var Modal = React.createClass({
+        componentDidMount: function() {
+          this.div = document.createElement('div');
+          document.body.appendChild(this.div);
+          this.componentDidUpdate();
+        },
+        componentDidUpdate: function() {
+          ReactDOM.render(<div>{this.props.children}</div>, this.div);
+        },
+        componentWillUnmount: function() {
+          var self = this;
+          // some async animation
+          setImmediate(function() {
```

Comment:
I don't see a problem with this? You guys should upgrade to jasmine2 so you can use the `done` feature of jasmine2 for async testing. Otherwise `waitsFor` is fine, just ugly API. You can also do `jest.runAllTimers()` to advance them manually – that way you can get rid of the `waitsFor`.

@zpao